api/script/index.js

```diff
@@ -564,68 +564,89 @@ module.exports = function transferFlow({utError: {fetchErrors}}) {
+                    pinCheckValueNew: params.pinCheckValueNew,
+                    mode: params.mode
+                }, {forward}).then(result => {
                     if (!result.issuerId) {
                         throw errors['transfer.unknownIssuer']();
                     }
+                    return {
+                        cardProductName: result.cardProductName,
+                        sourceAccount: result.sourceAccountNumber,
+                        sourceAccountName: result.sourceAccountName,
+                        destinationAccount: result.destinationAccountNumber,
+                        sourceCardProductId: result.sourceCardProductId,
+                        destinationAccountName: result.destinationAccountName,
+                        issuerId: result.issuerId,
+                        ledgerId: result.ledgerId,
+                        cardNumber: result.cardNumber,
+                        ordererId: result.ordererId
+                    };
+                });
+            });
 
-                    return result;
-                })
-                .catch(error => {
-                    params.abortAcquirer = error;
-                    return this.bus.importMethod('transferFlow.push.execute')(params, $meta);
-                })
-                .then(result => Object.assign(params, {
-                    cardProductName: result.cardProductName,
-                    sourceAccount: result.sourceAccountNumber,
-                    sourceAccountName: result.sourceAccountName,
-                    destinationAccount: result.destinationAccountNumber,
-                    sourceCardProductId: result.sourceCardProductId,
-                    destinationAccountName: result.destinationAccountName,
-                    issuerId: result.issuerId,
-                    ledgerId: result.ledgerId,
-                    cardNumber: result.cardNumber,
-                    ordererId: result.ordererId
-                }))
-                .then(result => !params.transferIdAcquirer && this.bus.importMethod(`db/${params.channelType}.terminal.nextId`)({
-                    channelId: result.channelId
-                }, {forward}))
-                .then(result => {
-                    if (params.transferIdAcquirer) {
-                        return params;
-                    }
-                    if (!result || !result[0] || !result[0][0] || !result[0][0].tsn) {
-                        throw errors['transfer.nextId']();
-                    }
-                    params.transferIdAcquirer = result[0][0].tsn;
-                    return params;
-                })
-                .then(params => this.bus.importMethod('transferFlow.push.execute')(params, $meta));
+            const processAcquirerError = Promise.resolve().then(() => {
+                if (!params.abortAcquirer) {
+                    return {};
+                }
+                return this.bus.importMethod('db/atm.card.processError')({
```

Comment:
Doesn't this result in an array?

Reply:
the result of this procedure is ignored as before the changes